base/permuteddimsarray.jl

```diff
@@ -253,6 +253,16 @@ end
     P
 end
 
+function Base._mapreduce_dim(f, op, init::Base._InitialValue, A::PermutedDimsArray, dims::Colon)
+    Base._mapreduce_dim(f, op, init, parent(A), dims)
+end
+
+function Base.mapreducedim!(f, op, B::AbstractArray{T,N}, A::PermutedDimsArray{T,N,P,Q}) where {T,N,P,Q}
+    C = PermutedDimsArray{T,N,Q,P,typeof(B)}(B)
```

Comment:
```suggestion
    C = PermutedDimsArray{T,N,P,Q,typeof(B)}(B)
```

Reply:
The inverse here is essential, here's a check: 
```julia
X = rand(10,20,300);
A = permutedims(X, (2,3,1)); summary(A)
P = PermutedDimsArray(X, (2,3,1)); summary(P)

sum(P, dims=2) ≈ sum(A, dims=2)

@btime sum(P, dims=2); # 56.125 μs
@btime sum(A, dims=3); # 15.292 μs

function Base.mapreducedim!(f, op, B::AbstractArray{T,N}, A::PermutedDimsArray{T,N,P,Q}) where {T,N,P,Q}
    C = PermutedDimsArray{T,N,Q,P,typeof(B)}(B)
    Base.mapreducedim!(f, op, C, parent(A))
    B
end

sum(P, dims=2) ≈ sum(A, dims=2)
@btime sum(P, dims=2); # 24.666 μs

function Base.mapreducedim!(f, op, B::AbstractArray{T,N}, A::PermutedDimsArray{T,N,P,Q}) where {T,N,P,Q}
    C = PermutedDimsArray{T,N,P,Q,typeof(B)}(B)  # with suggestion
    Base.mapreducedim!(f, op, C, parent(A))
    B
end

sum(P, dims=2) ≈ sum(A, dims=2) # DimensionMismatch
```